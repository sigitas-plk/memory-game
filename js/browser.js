import { prepCards } from "./game";

import * as s from "./reactivity";
const cards = {
  camera: "img/asset_camera.png",
  gempad: "img/asset_gamepad.png",
  glasses: "img/asset_glasses.png",
  headphones: "img/asset_headphones.png",
  joystick: "img/asset_joystick.png",
  keyboard: "img/asset_keyboard.png",
  monitor: "img/asset_monitor.png",
  mouse: "img/asset_mouse.png",
  notebook: "img/asset_notebook.png",
  phone: "img/asset_phone.png",
  speaker: "img/asset_speaker.png",
  usb: "img/asset_usb.png"
};

const state = {};
state.cards = prepCards(cards, 7);

export const createCard = (doc, card, index) => {
  const cardEl = doc.createElement("div");
  cardEl.classList.add("card");
  cardEl.dataset.index = index;
  const front = doc.createElement("div");
  front.classList.add("front");
  const back = doc.createElement("div");
  back.classList.add("back");
  back.style.backgroundImage = `url(${card.img})`;
  cardEl.appendChild(front);
  cardEl.appendChild(back);
  return cardEl;
};

export const renderCards = cards =>
  cards.forEach((c, i) =>
    document.getElementById("grid").appendChild(createCard(document, c, i))
  );

renderCards(state.cards);

console.log(state.cards);
