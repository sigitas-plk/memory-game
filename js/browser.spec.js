import { createCard } from "./browser";

describe("browser", () => {
  test("createCard should create dive nodes with front and back divs", () => {
    const card = {
      card: "test",
      img: "img/test.png",
      active: false
    };

    expect(createCard(document, card, 0)).toMatchSnapshot();
  });

  test("renderCards should render given cards in grid element", () => {});
});
