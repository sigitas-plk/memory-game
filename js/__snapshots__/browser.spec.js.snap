// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browser createCard should create dive nodes with front and back divs 1`] = `
<div
  class="card"
  data-index="0"
>
  <div
    class="front"
  />
  <div
    class="back"
    style="background-image: url(img/test.png);"
  />
</div>
`;
